src/NCronJob/Registry/JobDefinition.cs

```diff
@@ -120,5 +128,57 @@ public RecurringJobSchedule ToRecurringJobSchedule()
             TimeZone: TimeZone!);
     }
 
+    public void UpdateWith(JobOption? jobOption)
+    {
+        if (jobOption is null)
+        {
+            return;
+        }
+
+        if (jobOption.CronExpression is not null)
+        {
+            UserDefinedCronExpression = jobOption.CronExpression;
+            CronExpression = GetCronExpression(jobOption.CronExpression);
+
+            TimeZone = jobOption.TimeZoneInfo;
+        }
+
+        if (jobOption.Parameter is not null)
+        {
+            Parameter = jobOption.Parameter;
+        }
+
+        if (jobOption.ShouldCrashOnStartupFailure is not null)
+        {
+            ShouldCrashOnStartupFailure = jobOption.ShouldCrashOnStartupFailure;
+        }
+    }
+
+    private static CronExpression GetCronExpression(string expression)
+    {
+        var precisionRequired = DetermineAndValidatePrecision(expression);
```

Comment:
We could do the trimming here which would make #246 obsolete

Reply:
That would indeed be a good place to do that. 

Few options:
- We finish up #246 and I rebase this on top of it (which may be the fastest option).
- Your plate is full:
  - I embed your work from #246 in this PR
  - We clean this one first and deal with #246 then

Reply:
I don't think in #246 I can do move further without adding merge conflicts to your PR.
GIven that I can push the CRON creation further down.

If "I embed your work from" leads to no merge conflicts I would do this, otherwise ignore it for now.

Reply:
Either you rebase and solve conflicts or I rebase and solve conflicts ;-)
I'm fine with both ;-)

Reply:
If possible the latter - and we drop my PR. It is basically just about the test and the Trim.

Reply:
>If possible the latter - and we drop my PR. It is basically just about the test and the Trim.

Done